x/marker/types/msg.go

```diff
@@ -651,3 +652,44 @@ func (msg MsgSetAccountDataRequest) GetSigners() []sdk.AccAddress {
 	addr := sdk.MustAccAddressFromBech32(msg.Signer)
 	return []sdk.AccAddress{addr}
 }
+
+// NewMsgUpdateSendDenyListRequest creates a NewMsgUpdateSendDenyListRequest
+func NewMsgUpdateSendDenyListRequest(denom string, authority sdk.AccAddress, removeDenyAddresses, addDenyAddresses []string) *MsgUpdateSendDenyListRequest {
+	return &MsgUpdateSendDenyListRequest{
+		Denom:                 denom,
+		Authority:             authority.String(),
+		RemoveDeniedAddresses: removeDenyAddresses,
+		AddDeniedAddresses:    addDenyAddresses,
+	}
+}
+
+func (msg MsgUpdateSendDenyListRequest) ValidateBasic() error {
+	if err := sdk.ValidateDenom(msg.Denom); err != nil {
+		return err
+	}
+	if len(msg.AddDeniedAddresses) == 0 && len(msg.RemoveDeniedAddresses) == 0 {
+		return fmt.Errorf("both add and remove lists cannot be empty")
+	}
+
+	combined := []string{}
+	combined = append(combined, msg.AddDeniedAddresses...)
+	combined = append(combined, msg.RemoveDeniedAddresses...)
+	seen := make(map[string]bool)
+	for _, addr := range combined {
+		if _, err := sdk.AccAddressFromBech32(addr); err != nil {
+			return err
+		}
+		if seen[addr] {
+			return fmt.Errorf("denied address lists contain duplicate entries")
```

Comment:
```suggestion
			return fmt.Errorf("denied address lists contain duplicate entries: %q", addr)
```

Reply:
Never mind on this. Really, it'd be handy to list all offending addresses, but that could easily become unwieldy. It's probably okay to just make the simple statement and let the user identify the offending ones on their own.